src/common/base/src/base/cpu.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashSet;
+use std::fs::File;
+use std::io::Read;
+
+// Returns None if not get
+#[allow(unused_assignments)]
+pub fn get_physical_core_count() -> Option<usize> {
```

Comment:
Why not use `num_cpus::get_physical`?

Reply:
Oh， I did not know this function. But `num_cpus` also read the `/proc/cpuinfo` file.  Should I remove this to num_cpus ?

Reply:
Let's change it to num_cpus.